### PR TITLE
Add inventory advanced section

### LINKED ISSUE #34389: [Inventory] Advanced section
**User story**

As a merchant,
I want to control my product inventory
So that I can keep up with the demand

As a merchant,
I want to manage my inventory
So that my customers won’t purchase products that I don’t have at hand

**Design**
5sAIeTRd9Yp7nSCT33BAWz-fi-6498%3A284871

**Acceptance criteria**

- [ ] The `Allow customers to place backorders` checkbox allows customers to purchase a product when it is marked as out of stock (it’s the equivalent of the current `Allow` option under `Allow backorders?`).
- [ ] The tooltip next to `Allow customers to place backorders` reads: `Enable to let customers purchase this product when it is out of stock.`
- [ ] The `Show an ‘out of stock’ message` checkbox shows a note on the product page (it’s the equivalent of the current `Allow, but notify` option under `Allow backorders?`).
- [ ] The `Limit purchases to 1 item per order` checkbox removes the quantity box on the product page so that customers can only purchase 1 item.
- [ ] The tooltip next to `Limit purchases to 1 item per order` reads: `Enable this to only allow one of this item to be bought in a single order.`

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds the advanced section for product inventory management.  Note that some styling changes around radios and radio labels exist in https://github.com/woocommerce/woocommerce/pull/35047.
Screen Shot 2022-10-18 at 3 46 51 PM

Closes #34389  .

### How to test the changes in this Pull Request:

1. Enable/disable the stock management for products at `wp-admin/admin.php?page=wc-settings&tab=products&section=inventory`
2. Under the product page (Products -> Add new (MVP)), click "Advanced" in the inventory section
3. Note that when stock management is enabled, the "out of stock" options exist
4. Note that in both cases, the limit 1 per purchase checkbox exists

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by louwie17 — DISMISSED** [review]
This tested well, and changes are simple and straightforward, nice work!
It would be great to add a couple tests around this though.

Approving this, there are some merge conflicts, so let me know when you need a re-approval.

**Comment by joshuatf** [comment]
Thanks, @louwie17!  Rebased and also added tests in https://github.com/woocommerce/woocommerce/pull/35164/commits/e121f1a8d6fb1700b0500dad8487ed3ed911cbf9

File: plugins/woocommerce-admin/client/products/sections/product-inventory-section/test/product-inventory-section.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			</Form>
+		);
+
+		expect(
+			screen.getByText( 'Track quantity for this product' )
+				.previousSibling
+		).toHaveClass( 'is-disabled' );
+	} );
+
+	it( 'should not render the manage stock section if inventory management is turned on', () => {
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/sections/product-inventory-section/test/product-inventory-section.spec.tsx]: Should this actually be that the manage stock is not disabled?

**Review by louwie17 — COMMENTED** [review]
This is looking good @joshuatf, also appreciate you adding tests.
I did leave one comment in one of the tests, the name of it kinda confused me, in relation to what the test was actually doing.
Also when the track quantity is enabled it seems the margin is a lot bigger between the `Advanced ^` button and the `When out of stock` label, compared to the margin between `Restrictions` and the toggle when track quantity is disabled.
See screenshot:
Screen Shot 2022-10-27 at 2 52 45 PM

File: plugins/woocommerce-admin/client/products/sections/product-inventory-section/test/product-inventory-section.spec.tsx (hunk 1 above)
Comment (joshuatf) [reply plugins/woocommerce-admin/client/products/sections/product-inventory-section/test/product-inventory-section.spec.tsx]: It should!  Thanks for the catch!

**Comment by joshuatf** [comment]
Thanks for the review, @louwie17.  I must have screwed up the rebase and created the margin issue.  Should be resolved now in https://github.com/woocommerce/woocommerce/pull/35164/commits/5db6f0707f0f7bcc6efba546c9b2afe12d6ddf47.

**Review by louwie17 — DISMISSED** [review]
Styling looks good now, nice work @joshuatf , this looks good!

Btw you will likely have to rebase again with latest of trunk to fix the code sniff error.

**Review by louwie17 — APPROVED** [review]
LGTM 🚀

**Pull request merged**